pkgs/os-specific/linux/ffado/default.nix

```diff
@@ -1,44 +1,57 @@
 { stdenv, fetchurl, scons, pkgconfig, which, makeWrapper, python
-, expat, libraw1394, libconfig, libavc1394, libiec61883, libxmlxx, glibmm
+, expat, libraw1394, libconfig, libavc1394, libiec61883, libxmlxx
+, alsaLib, dbus, dbus_cplusplus
 
 # Optional dependencies
-, libjack2 ? null, dbus ? null, dbus_cplusplus ? null, alsaLib ? null
 , pyqt4 ? null, dbus-python ? null, xdg_utils ? null
 
 # Other Flags
-, prefix ? ""
+, libOnly ? false
 }:
 
 let
 
   shouldUsePkg = pkg: if pkg != null && pkg.meta.available then pkg else null;
 
-  libOnly = prefix == "lib";
-
-  optLibjack2 = shouldUsePkg libjack2;
-  optDbus = shouldUsePkg dbus;
-  optDbus_cplusplus = shouldUsePkg dbus_cplusplus;
-  optAlsaLib = shouldUsePkg alsaLib;
+# Mixer/tools dependencies
+# FIXME: can we build mixer separately? Then we can farewell `libOnly` mode
+# (otherwise we have perfect loop via qt -> pulseaudio -> jack -> ffado -> qt)
```

Comment:
We can probably override ffado in jack's callPackage-jack shouldn't need Qt.

Reply:
ffado' mixer is actually separate app, which have completely own dependencies (dbus-python and PyQt), so I feel  separate expression may be better solution. But would be nice to hear someone who use ffado.